clippy_lints/src/methods/mod.rs

```diff
@@ -783,63 +784,42 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
             return;
         }
 
+        let (method_names, arg_lists) = method_calls(expr, 2);
+        let method_names: Vec<LocalInternedString> = method_names.iter().map(|s| s.as_str()).collect();
+        let mut method_names = method_names.iter().map(|s| s.as_ref()).chain(iter::repeat(""));
+
+        match [method_names.next().unwrap(), method_names.next().unwrap()] {
```

Comment:
Slice patterns could be used instead of `chain(repeat(""))` trick.

```rust
let method_names: Vec<_> = method_names.iter().map(|s| s.as_ref()).collect();

match method_names.as_slice() {
    ["unwrap", "get"] => ...,
    ...
    ["map_or"] => ...,
    ...
}
```

Reply:
Hmm, does `["map_or", ..]` work?